enclave_wrangler/dataset_download_new_api.py

```diff
@@ -12,67 +13,116 @@
+        "https://unite.nih.gov/ontology-metadata/api/ontology/linkTypesForObjectTypes" --data '{
+            "objectTypeVersions": {
+                "ri.ontology.main.object-type.a11d04a3-601a-45a9-9bc2-5d0e77dd512e": \
+                "00000001-9834-2acf-8327-ecb491e69b5c"
+            }
+        }'
+        jq '..|objects|.apiName//empty' $@
+        """
+        # TODO: @Siggie I tried using the above curl in Python but I got this (- Joe 2022/08/21):
+        #  {'errorCode': 'INVALID_ARGUMENT', 'errorName': 'Conjure:UnsupportedMediaType', 'errorInstanceId':
+        #  '7976c277-8187-4a5a-91b9-2e8bd1c9934c', 'parameters': {}}
+        data = {
+            "objectTypeVersions": {
+                "ri.ontology.main.object-type.a11d04a3-601a-45a9-9bc2-5d0e77dd512e":
+                    "00000001-9834-2acf-8327-ecb491e69b5c"
+            }
+        }
+        url = f'{self.base_url}/ontology-metadata/api/ontology/linkTypesForObjectTypes'
+        response = requests.post(url, headers=self.headers, data=data)
+        response_json = response.json()
+        return response_json
+
+
+def run(request_types: List[str]) -> Dict[str, Dict]:
     """Run"""
-    types = objTypes()
-    print('\n'.join([t['apiName'] for t in types]))
-
+    client = EnclaveClient()
+    request_funcs: Dict[str, Callable] = {
+        'objects': client.objects,
+        'object_types': client.obj_types,
+        'link_types': client.link_types,
+    }
+    results = {}
+    for req in request_types:
+        results[req] = request_funcs[req]()
+        # if req == 'object_types':
+        #     print('\n'.join([t['apiName'] for t in results[req]['types']]))
+    return results
 
 
-def get_parser():
-    """Add required fields to parser.
-
-    Returns:
-        ArgumentParser: Argeparse object.
-    """
+def cli():
+    """Command line interface for package."""
     package_description = 'Tool for working w/ the Palantir Foundry enclave API. ' \
-          'This part is for downloading enclave datasets.'
+                          'This part is for downloading enclave datasets.'
     parser = ArgumentParser(description=package_description)
 
+    # parser.add_argument(
+    #     '-a', '--auth_token_env_var',
+    #     default='PALANTIR_ENCLAVE_AUTHENTICATION_BEARER_TOKEN',
+    #     help='Name of the environment variable holding the auth token you want to use')
+    # todo: 'objects' alone doesn't make sense because requires param `object_type`
     parser.add_argument(
-        '-a', '--auth_token_env_var',
-        default='PALANTIR_ENCLAVE_AUTHENTICATION_BEARER_TOKEN',
-        help='Name of the environment variable holding the auth token you want to use')
-
-    return parser
-
-def cli():
-    """Command line interface for package.
-
-    Side Effects: Executes program."""
-    parser = get_parser()
+        '-r', '--request-types',
+        nargs='+', default=['object_types', 'objects', 'link_types'],
```

Comment:
Repurposed the CLI like this for now. But I need to add parameterization for 'objects'.

Can use like: `python enclave_wrangler/dataset_download_new_api.py -r object_types`